### PR TITLE
AppCompat: converting two HWR properties to no-op.

### LINKED ISSUE #20043: HttpWebRequest.ReadWriteTimeout property returns NotImplemented
Looks like HttpWebRequest.ReadWriteTimeout throws a NotImplemented exception.
Also HttpWebRequest.ConnectionGroupName

Expected:
Either work with expected behavior, or remove from NET Core altogether as this will result in an unexpected runtime error.

### PR DESCRIPTION
**Pull request by CIPop**
Fixes #20043

### CODE REVIEW

**Comment by CIPop** [comment]
@dotnet-bot help

**Comment by CIPop** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netfx Windows_NT Debug

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 1]
```diff
@@ -500,17 +500,9 @@ public virtual bool AllowAutoRedirect
...
-            {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-            set
-            {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-        }
+        // .Net Framework behavior difference: ConnectionGroupName cannot be changed.
+        // For app-compat reasons we allow apps to change it.
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Nit: I don't think we need the comment.  The first line is essentially referring to what, after this is checked in, will have been an old implementation that never shipped and won't actually exist anywhere.  Same with the other occurrence of it below.

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 2]
```diff
@@ -795,18 +787,9 @@ public Version ProtocolVersion
...
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-            set
-            {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-        }
+        // .Net Framework behavior difference: ReadWriteTimeout cannot be changed.
+        // For app-compat reasons we allow apps to change it and return the .Net Framework default.
+        public int ReadWriteTimeout { get; set; } = 300000;
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Nit: desktop has a named const for the value; consider doing that here:
https://referencesource.microsoft.com/#System/net/System/Net/HttpWebRequest.cs,115

**Comment by CIPop** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netfx Windows_NT Debug

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 3]
```diff
@@ -795,18 +786,7 @@ public Version ProtocolVersion
...
-            get
-            {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-            set
-            {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
-            }
-        }
+        public int ReadWriteTimeout { get; set; } = DefaultReadWriteTimeout;
```
Comment (justinvp) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Should the setter still do parameter validation? E.g. in desktop it throws if `value <= 0 && value != System.Threading.Timeout.Infinite`.

https://github.com/Microsoft/referencesource/blob/90b323fe52bec428fe4bd5f007e9ead6b265d553/System/net/System/Net/HttpWebRequest.cs#L1038-L1040

Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Yes, we should do whatever validation desktop does, within reason.

Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Good point.

**Comment by CIPop** [comment]
@mmitche Is this something known? I've never seen Crosscheck fail before:

```
16:22:03 2017-05-25 16:22:03 INFO Retrieving 2 files for pull request 20332
16:22:03 2017-05-25 16:22:03 INFO Scoring pull request
16:22:03 2017-05-25 16:22:03 ERROR Critical exception happend. Details dumped into cross_exception.log
16:22:03 
16:22:03 D:\j\workspace\CROSS_check_p---251af28f\cross-tool>exit 0 
```

**Comment by mmitche** [comment]
@buckaroo-banzai @PavelBansky look familiar?

**Comment by mmitche** [comment]
It had an odd sort of exception.  I have no idea whether this means the cross log wasn't there, or it didn't find.  Never seen this error before.

16:22:03 Run condition [Current build status] enabling perform for step [[Archive the artifacts]]
16:22:03 Archiving artifacts
16:22:05 ERROR: No artifacts found that match the file pattern "**/cross-log.json". Configuration error?
16:22:05 ERROR: java.lang.InterruptedException: no matches found within 10000
16:22:05 Build step 'Flexible publish' changed build result to FAILURE

**Comment by CIPop** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netfx Windows_NT Debug

**Comment by CIPop** [comment]
Timing issues with one of the tests:

```
Assert.Throws() Failure\r\nExpected: typeof(System.InvalidOperationException)\r\nActual:   (No exception was thrown)
Stack Trace :
   at System.Net.Tests.HttpWebRequestTest.ReadWriteTimeout_RequestSubmitted_Fail()
```

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 4]
```diff
@@ -795,16 +788,25 @@ public Version ProtocolVersion
...
             {
-                throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException);
+                if (RequestSubmitted)
+                {
+                    throw new InvalidOperationException(SR.net_reqsubmitted);
+                }
+
+                if (value <= 0 && value != System.Threading.Timeout.Infinite)
+                {
+                    throw new ArgumentOutOfRangeException("value", SR.net_io_timeout_use_gt_zero);
```
Comment (justinvp) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Nit: `nameof(value)`

**Comment by CIPop** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netfx Windows_NT Debug

**Comment by stephentoub** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Debug x64 please
@dotnet-bot test OuterLoop netfx Windows_NT Debug x64 please

**Comment by stephentoub** [comment]
@danmosemsft:
System.Diagnostics.Tests.ProcessStartInfoTests.StartInfo_TextFile_ShellExecute [FAIL]
        System.NullReferenceException : Object reference not set to an instance of an object.
        Stack Trace:
           D:\j\workspace\outerloop_net---92aeb271\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs(1013,0): at System.Diagnostics.Tests.ProcessStartInfoTests.StartInfo_TextFile_ShellExecute()

**Pull request merged**